Mathlib/GroupTheory/SpecificGroups/Dihedral.lean

```diff
@@ -204,4 +204,43 @@ theorem exponent : Monoid.exponent (DihedralGroup n) = lcm n 2 := by
       exact (orderOf_sr 0).symm
 #align dihedral_group.exponent DihedralGroup.exponent
 
+/-- If n is odd, then the Dihedral group of order 2n has n*(n+3) pairs of commuting elements. -/
+def OddCommuteEquiv (hn : Odd n) : { p : DihedralGroup n × DihedralGroup n // Commute p.1 p.2 } ≃
+    ZMod n ⊕ ZMod n ⊕ ZMod n ⊕ ZMod n × ZMod n :=
+  let u := ZMod.unitOfCoprime 2 (Nat.prime_two.coprime_iff_not_dvd.mpr hn.not_two_dvd_nat)
+  have hu : ∀ a : ZMod n, a + a = 0 ↔ a = 0 := fun a => ZMod.add_self_eq_zero_iff_eq_zero hn
+  { toFun := fun p => match p.1.1, p.1.2 with
+      | sr i, r _ => Sum.inl i
+      | r _, sr j => Sum.inr (Sum.inl j)
+      | sr i, sr j => Sum.inr (Sum.inr (Sum.inl (i + j)))
+      | r i, r j => Sum.inr (Sum.inr (Sum.inr ⟨i, j⟩))
+    invFun := fun p => match p with
```

Comment:
```suggestion
    invFun := fun
```
`fun` is `match` too!

Reply:
Oh, this is great! I suspected that there was probably an easier way :)